extensions/iceberg/src/test/java/io/deephaven/iceberg/junit5/CatalogAdapterBase.java

```diff
@@ -0,0 +1,28 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.junit5;
+
```

Comment:
Can you please use Junit4 instead of Junit5, because for my case, I will be writing deephaven tables to iceberg, and for creating these tables, I would need an execution context which so far I have been using with Junit4 using
```
    @Rule
    public final EngineCleanup framework = new EngineCleanup();
```
Let me know if I am missing something.

Reply:
It should be super simple to create JUnit 5 tests using the same logic;  I don't think it's unreasonable to do this:

```java
    private EngineCleanup engineCleanup = new EngineCleanup();

    @BeforeEach
    void setUp() throws Exception {
        engineCleanup.setUp();
    }

    @AfterEach
    void tearDown() throws Exception {
        engineCleanup.tearDown();
    }
}
```

We could create a JUnit 5 extension to make it just as easy as `@Rule` (I think I've done this on some branch, but I forget where...).

I really really really think we should be preferring JUnit 5 for new tests going forward.

I've updated `io.deephaven.iceberg.junit5.CatalogAdapterBase` to use `EngineCleanup` in anticipation of the ticking testing that should be upcoming.

Reply:
Makes sense, thanks!